acceptance/tests/puppet_device_test.rb

```diff
@@ -1,74 +1,73 @@
   on(master, "puppet agent -t --server #{master}")
 end
 
-# set sensible defaults
-on(master, "cp /etc/puppetlabs/puppet/puppet.conf /etc/puppetlabs/puppet/puppet.conf.bak")
-on(master, "puppet config set server $(hostname --fqdn)")
-
-codedir = "#{master.puppet['codedir']}/modules"
-confdir = master.puppet['confdir']
-common_options="--config /tmp/puppet.conf --debug --trace -w0"
-
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-test_device"))
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-device_manager"))
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-resource_api"))
-
-apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
-# create and configure test user to trigger PUP-9642
-user {
-  'test':
-    ensure => 'present',
-}
-
-# testfile for --apply
-file {
-  '/tmp/spinner.pp':
-    content => "spinner { '999': ensure => 'absent' }"
-}
-
-# install and configure the Resource API
-service { 'puppetserver': }
-include 'resource_api::server', 'resource_api::agent'
+with_puppet_running_on(master, {}) do
+  on(master, "puppet config set server $(hostname --fqdn)")
+
+  codedir = "#{master.puppet['codedir']}/modules"
+  confdir = master.puppet['confdir']
+  common_options="--debug --trace -w0"
+
+  on(master, puppet("module install --target-dir #{codedir} puppetlabs-test_device"))
+  on(master, puppet("module install --target-dir #{codedir} puppetlabs-device_manager"))
+  on(master, puppet("module install --target-dir #{codedir} puppetlabs-resource_api"))
+
+  apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
+  # create and configure test user to trigger PUP-9642
+  user {
+    'test':
+      ensure => 'present',
+  }
+
+  # testfile for --apply
+  file {
+    '/tmp/spinner.pp':
+      content => "spinner { '999': ensure => 'absent' }"
+  }
+
+  # install and configure the Resource API
+  include 'resource_api::server', 'resource_api::agent'
 MANIFEST
 
-on(master, "cp -v #{confdir}/puppet.conf /tmp/puppet.conf")
-on(master, "echo \"#{<<-CONF}\" >> /tmp/puppet.conf")
-[main]
-user = test
+  on(master, "echo \"#{<<-CONF}\" >> #{confdir}/puppet.conf")
+  [main]
+  user = test
 CONF
 
-# configure device credentials
-on(master, puppet("apply -e 'device_manager { [\"spinny1.example.com\", \"spinny2.example.com\", \"spinny3.example.com\"]: type => 'spinner', credentials => { facts_cpu_time => 1 }, include_module => false }'"))
+  # configure device credentials
+  on(master, puppet("apply -e 'device_manager { [\"spinny1.example.com\", \"spinny2.example.com\", \"spinny3.example.com\"]: type => 'spinner', credentials => { facts_cpu_time => 1 }, include_module => false }'"))
 
-# run operation without certificate
-on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com --facts")
+  # run operation without certificate
+  on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com --facts")
```

Comment:
Is this where we start interacting with the master? If so we might want to move the bits above out of the `with_puppet..` block

Reply:
Fair enough.